src/develop/imageop.h

```diff
@@ -742,18 +747,24 @@ char *dt_iop_set_description(dt_iop_module_t *module, const char *main_text,
 static inline dt_iop_gui_data_t *_iop_gui_alloc(dt_iop_module_t *module, size_t size)
 {
   module->gui_data = (dt_iop_gui_data_t*)calloc(1, size);
-  dt_pthread_mutex_init(&module->gui_lock,NULL);
   return module->gui_data;
 }
 #define IOP_GUI_ALLOC(module) \
   (dt_iop_##module##_gui_data_t *)_iop_gui_alloc(self,sizeof(dt_iop_##module##_gui_data_t))
 
-#define IOP_GUI_FREE \
-  dt_pthread_mutex_destroy(&self->gui_lock);if(self->gui_data){free(self->gui_data);} self->gui_data = NULL
+#define IOP_GUI_FREE                                                                                              \
+  do                                                                                                              \
+  {                                                                                                               \
+    free(self->gui_data);                                                                                         \
+    self->gui_data = NULL;                                                                                        \
+  } while(0)
```

Comment:
I understand you added the do while(0) to make the define work properly after if. In this particular case though, IOP_GUI_FREE should never be called within an if statement anyway. But I guess it doesn't hurt...